lib/AST/NameLookup.cpp

```diff
@@ -137,14 +137,40 @@ void DebuggerClient::anchor() {}
 void AccessFilteringDeclConsumer::foundDecl(
     ValueDecl *D, DeclVisibilityKind reason,
     DynamicLookupInfo dynamicLookupInfo) {
-  if (D->hasInterfaceType() && D->isInvalid())
-    return;
   if (!D->isAccessibleFrom(DC))
     return;
 
   ChainedConsumer.foundDecl(D, reason, dynamicLookupInfo);
 }
 
+void UsableFilteringDeclConsumer::foundDecl(ValueDecl *D,
+    DeclVisibilityKind reason, DynamicLookupInfo dynamicLookupInfo) {
```

Comment:
Could you add a fast path for decls from other `SourceFile`?
Comparing them using `SourceLoc`/`SouceRange` doesn't look right.

Reply:
Good point, thanks

Reply:
Thinking about this more... `rangeContainsTokenLoc` is valid even if the locations are in different buffers. It ends up just comparing the underlying values, ie. start <= loc <= end. That would just be false if they were in different buffers.

I could add a check for the locations having the same buffer id, but the way that's retrieved is through iterating over all the buffers and checking their start anyway. Ie. it's assuming the same thing.

In chatting to you about this you brought up the possibility of the range not being valid though, so I will check for that :)

Reply:
> I could add a check for the locations having the same buffer id, but the way that's retrieved is through iterating over all the buffers and checking their start anyway.

For future reference, checking same source file can be done by something like:
```cpp
auto declSF = VD->getDeclContext()->getParentSouceFile();
auto sourceFileRange = SM.getRangeForBuffer(declSF->getBufferID());
return SM.rangeContainsTokenLoc(souceFileRange, Loc);
```